### PR TITLE
40-0247 - Write automated tests

### PR DESCRIPTION
**Pull request by tlei123**
\[Making this Ready for Review in case we decide to merge as is, and open follow-up tickets for the outstanding file-upload page\]

## Summary

- Write automated tests for Pres. Mem. Cert. (40-0247)
- Team: Veteran Facing Forms
- No Flipper yet; form's still in-development

## Related issue(s)

- _Link to ticket created in va.gov-team-forms repo_
department-of-veterans-affairs/va.gov-team-forms#481

## Testing done

- All tests passed locally

## What areas of the site does it impact?

This form ONLY

## Requested Feedback

Beyond the basic code-review, I also need guidance on how to fix the coverage-report.  Even though all unit-tests were in-place for all pages and all passed when I ran `yarn test:unit src/applications/simple-forms/40-0247/tests/unit/**/*.unit.spec.jsx`, the Pages coverage sub-report's showing some confusing results, as if the v3-web-components pages' unit-tests, and the non-v3 file-upload page's, hadn't run at all.  \[See screenshot below for the Pages sub-report\]

image

### CODE REVIEW

File: src/applications/simple-forms/shared/tests/e2e/helpers.js [hunk 1]
```diff
@@ -1,3 +1,19 @@
+export const getPagePaths = formConfig => {
```
Comment (tlei123) [inline src/applications/simple-forms/shared/tests/e2e/helpers.js:1]: This'll save us from manually writing out page-paths for v3-web-component unit-tests.

**Review by rhasselle-oddball — COMMENTED** [review]
changes overall look good. yea as for the code coverage thing -
code coverage only looks for things like functions, lambdas, statements with conditionals. so it doesn't cover simple constants of schemas. So its entirely possible to have entire files with no branching logic, and thus there's nothing for code coverage to check.

I wonder if it was defined as a const, and then exported, vs just directly exported, then it might count toward code coverage?

File: src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js [hunk 2]
```diff
@@ -1,25 +1,23 @@
 import React from 'react';
 
-import {
-  // titleSchema,
-  titleUI,
-} from 'platform/forms-system/src/js/web-component-patterns';
-import {
-  addressNoMilitaryUI,
-  addressNoMilitarySchema,
-} from 'platform/forms-system/src/js/web-component-patterns/addressPattern.jsx';
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js:10]: It's 'cuz we have a non-v3 qty-field below the address stuff, so i can't mix the v3 address in there.

**Comment by tlei123** [comment]
> I wonder if it was defined as a const, and then exported, vs just directly exported, then it might count toward code coverage?

@rhasselle-oddball , I just double-checked my page-files and they're all exported directly, no const.

File: src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js:10]: ok, i'm gonna change the whole page to v3 now that i can copy-n-paste you v3 number-patterns.  i'll keep you posted also on my coverage-weirdness.

**Comment by tlei123** [comment]
@rhasselle-oddball , the _aggregate_ coverage-percentages are looking good \[see screenshot below\].  The file-upload page is still WIP right now.  I only have our 2 regular fields/errors tests in-place for this page, but I already have a checklist-item for it in my follow-up ticket [#585](https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/585).

Let me know if the aggregate numbers are sufficient to for approval.  Thanks much!

image

**Review by rhasselle-oddball — APPROVED** [review]
lots of tests! nice

**Pull request merged**